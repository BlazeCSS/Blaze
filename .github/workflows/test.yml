name: CI

on:
  push:
    branches:
      - master

jobs:
  build:
    name: Install, test and build
    
    runs-on: ubuntu-latest
    
    steps:
      - name: Checkout repo
        uses: actions/checkout@master

      - name: Install dependencies
        run: npm ci

      - name: Test
        run: npm run test

      - name: Build site
        run: npm run build
      
      - name: Switch to site dir
        run: cd ./packages/site

      - name: Deploy site
        uses: w9jds/firebase-action@master
        with:
          args: deploy --only hosting
        env:
          FIREBASE_TOKEN: ${{ secrets.FIREBASE_TOKEN }}
          PROJECT_ID: "default"
