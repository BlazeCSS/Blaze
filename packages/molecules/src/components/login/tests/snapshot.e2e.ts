import { snapIt } from '../../../test';

const component = 'login';

describe(component, () => {
  const snap = snapIt(component);

  describe('renders', () => {
    snap('correctly', '<blaze-login></blaze-login>');

    snap(
      'all settings',
      `<blaze-login username-label="Email address: "
                    username-hint="The email address you used when you signed up"
                    username-type="email"
                    username-autocomplete="@me.com"
                    password-label="Password: "
                    password-hint="It's probably Pa55w0rd..."
                    remember-me-label="Remember me"
                    remember-me>
      </blaze-login>`
    );
  });
});
