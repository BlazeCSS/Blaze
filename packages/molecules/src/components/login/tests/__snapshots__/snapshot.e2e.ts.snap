// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login renders all settings 1`] = `"<blaze-login username-label=\\"Email address: \\" username-hint=\\"The email address you used when you signed up\\" username-type=\\"email\\" username-autocomplete=\\"@me.com\\" password-label=\\"Password: \\" password-hint=\\"It's probably Pa55w0rd...\\" remember-me-label=\\"Remember me\\" remember-me=\\"\\" class=\\"hydrated\\"><fieldset class=\\"o-fieldset c-list c-list--inline c-list--unstyled\\"><label class=\\"c-label o-form-element\\">Email address: <input type=\\"email\\" class=\\"c-field undefined c-field--label username\\"><div role=\\"tooltip\\" class=\\"c-hint\\">The email address you used when you signed up</div></label><label class=\\"c-label o-form-element\\">Password: <input type=\\"password\\" class=\\"c-field undefined c-field--label password\\"><div role=\\"tooltip\\" class=\\"c-hint\\">It's probably Pa55w0rd...</div></label><div class=\\"o-form-element\\"><blaze-toggle type=\\"brand\\" class=\\"hydrated\\"><label class=\\"c-toggle c-toggle--brand\\"><input type=\\"checkbox\\" aria-checked=\\"false\\"><div class=\\"c-toggle__track\\"><div class=\\"c-toggle__handle\\"></div></div>Remember me</label></blaze-toggle></div></fieldset></blaze-login>"`;

exports[`login renders correctly 1`] = `"<blaze-login class=\\"hydrated\\"><fieldset class=\\"o-fieldset c-list c-list--inline c-list--unstyled\\"><label class=\\"c-label o-form-element\\">Username<input type=\\"text\\" class=\\"c-field undefined c-field--label username\\"></label><label class=\\"c-label o-form-element\\">Password<input type=\\"password\\" class=\\"c-field undefined c-field--label password\\"></label></fieldset></blaze-login>"`;
